src/pybind/mgr/dashboard/controllers/_crud.py

```diff
@@ -286,21 +308,38 @@ class CRUDResourceMethod(NamedTuple):
     doc: EndpointDoc
 
 
-class CRUDEndpoint(NamedTuple):
-    router: APIRouter
-    doc: APIDoc
-    set_column: Optional[Dict[str, Dict[str, str]]] = None
-    actions: List[TableAction] = []
-    permissions: List[str] = []
-    forms: List[Form] = []
-    meta: CRUDMeta = CRUDMeta()
-    get_all: Optional[CRUDCollectionMethod] = None
-    create: Optional[CRUDCollectionMethod] = None
-
+class CRUDEndpoint:
     # for testing purposes
     CRUDClass: Optional[RESTController] = None
     CRUDClassMetadata: Optional[RESTController] = None
-    # ---------------------
+
+    # pylint: disable=R0902
+    def __init__(self, router: APIRouter, doc: APIDoc,
+
+                 set_column: Optional[Dict[str, Dict[str, str]]] = None,
+                 actions: Optional[List[TableAction]] = None,
+                 permissions: Optional[List[str]] = None, forms: Optional[List[Form]] = None,
+                 meta: CRUDMeta = CRUDMeta(), get_all: Optional[CRUDCollectionMethod] = None,
+                 create: Optional[CRUDCollectionMethod] = None):
+        self.router = router
+        self.doc = doc
+        self.set_column = set_column
+        if actions:
+            self.actions = actions
+        else:
+            self.actions = []
+
+        if forms:
+            self.forms = forms
+        else:
+            self.forms = []
+        self.meta = meta
+        self.get_all = get_all
+        self.create = create
+        if permissions:
+            self.permissions = permissions
+        else:
+            self.permissions = []
```

Comment:
Aren't we using a Python version that supports dataclasses yet (>=3.7)? If I'm not wrong this should be totally unnecessary with dataclasses (and if not supported yet, there's a backport dataclasses library for 3.6).

Reply:
We can add dataclasses in another pr. But it seems like it would be overkill to add dataclasses when a simple __init__ is working fine.

Reply:
@pereman2 are you working on this one, or for a later one?

Reply:
[I opened a tracker](https://tracker.ceph.com/issues/58716) for this since this will require another sprint.